packages/agreement-process/test/agreement.integration.test.ts

```diff
@@ -0,0 +1,738 @@
+      );
+
+      await writeInReadmodel<EService>(eservice, eservices);
+
+      const authData = getRandomAuthData(consumer.id);
+      const apiAgreementPayload: ApiAgreementPayload = {
+        eserviceId: eservice.id,
+        descriptorId: descriptor.id,
+      };
+
+      await expect(() =>
+        agreementService.createAgreement(apiAgreementPayload, authData)
+      ).rejects.toThrowError(tenantIdNotFound(consumer.id));
+    });
+
+    it("should throw a missingCertifiedAttributesError error when the EService producer and Agreement consumer are different Tenants, and the consumer is missing a Descriptor certified Attribute", async () => {
+      const eserviceProducer: Tenant = buildTenant();
+
+      // Descriptor has two certified attributes
+      const certifiedDescriptorAttribute1: EServiceAttribute =
+        buildEServiceAttribute();
+      const certifiedDescriptorAttribute2: EServiceAttribute =
+        buildEServiceAttribute();
+
+      const descriptor = {
+        ...buildDescriptorPublished(generateId<DescriptorId>()),
+        attributes: {
+          certified: [
+            [certifiedDescriptorAttribute1],
+            [certifiedDescriptorAttribute2],
+          ],
+          verified: [],
+          declared: [],
+        },
+      };
+
+      // In this case, the consumer is missing one of the two certified attributes
+      const certifiedTenantAttribute1: TenantAttribute =
+        buildCertifiedTenantAttribute(certifiedDescriptorAttribute1.id);
+
+      const consumer = {
+        ...buildTenant(),
+        attributes: [certifiedTenantAttribute1],
+      };
+
+      const eservice = buildEService(
+        generateId<EServiceId>(),
+        eserviceProducer.id,
+        [descriptor]
+      );
+
+      await writeInReadmodel<Tenant>(eserviceProducer, tenants);
+      await writeInReadmodel<Tenant>(consumer, tenants);
+      await writeInReadmodel<EService>(eservice, eservices);
+
+      const authData = getRandomAuthData(consumer.id);
+      const apiAgreementPayload: ApiAgreementPayload = {
+        eserviceId: eservice.id,
+        descriptorId: eservice.descriptors[0].id,
+      };
+
+      await expect(
+        agreementService.createAgreement(apiAgreementPayload, authData)
+      ).rejects.toThrowError(
+        missingCertifiedAttributesError(descriptor.id, consumer.id)
+      );
+    });
+
+    it("should throw a missingCertifiedAttributesError error when the EService producer and Agreement consumer are different Tenants, and the consumer has a Descriptor certified Attribute revoked", async () => {
+      const eserviceProducer: Tenant = buildTenant();
+
+      // Descriptor has two certified attributes
+      const certifiedDescriptorAttribute1: EServiceAttribute =
+        buildEServiceAttribute();
+      const certifiedDescriptorAttribute2: EServiceAttribute =
+        buildEServiceAttribute();
+
+      const descriptor: Descriptor = {
+        ...buildDescriptorPublished(),
+        attributes: {
```

Comment:
Same here, is there a reason why you are not passing the attributes as params to `buildDescriptorPublished`